Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/TiersRepository.cs

```diff
@@ -38,6 +44,11 @@ public async Task<DbPaginationResult<Tier>> FindAll(PaginationFilter paginationF
         return paginationResult;
     }
 
+    public async Task<Tier> FindById(TierId tierId, CancellationToken cancellationToken)
+    {
+        return await _tiersDbSet.Include("Identities").SingleAsync(t => t.Id == tierId, cancellationToken);
```

Comment:
`SingleAsync` throws an exception, which leads to an http status code 500... But we want a 404. So return null and throw a NotFoundException in the command handler